### PR TITLE
Stubs out the GSC Overview and Proposals pages

### PR DESCRIPTION
**Pull request by sentilesdal**
just getting some basic stubbing in.  neither link will show in the sidebar if you don't have the flag in localstorage

image

image

### CODE REVIEW

**Comment by sentilesdal** [comment]
#### 📝 Changed routes:


| [**/gsc-overview**](https://elf-council-frontend-staging-2wessao57-element-finance.vercel.app/gsc-overview) | [**/gsc-proposals**](https://elf-council-frontend-staging-2wessao57-element-finance.vercel.app/gsc-proposals) |
|:-:|:-:|
|<a href="https://elf-council-frontend-staging-2wessao57-element-finance.vercel.app/gsc-overview">Screenshot of /gsc-overview</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/fBOaxQllvFP_bLPJ23sc0at5p6fOQSELRsdQYiW0rnqijXMxq0Gu6qgpJswCD_k4ubV0w7yGjmpZwOg5QK2vyA.png">(view full size)</a> |<a href="https://elf-council-frontend-staging-2wessao57-element-finance.vercel.app/gsc-proposals">Screenshot of /gsc-proposals</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/NaCd-oIEdqtxnfq7RnhaK-LaNNQf2GeMmD58g0wVvd9crVIpZmQ3MzLRUIb5aoWvF0U3VJNA1zIZmuvnqdZaww.png">(view full size)</a> |



| [**/proposals**](https://elf-council-frontend-staging-2wessao57-element-finance.vercel.app/proposals) |
|:-:|
|<a href="https://elf-council-frontend-staging-2wessao57-element-finance.vercel.app/proposals">Screenshot of /proposals</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/BFspE--ljW-zyW5i28h289Km0LWMfYe9dfcaQrAIIYKCZ1aN-3fbwITtZC3cYG6PKx7elH-DQmSh18N9kAd5cA.png">(view full size)</a> |







Commit 91c953099d4f0741476f1c1464fd1027f85498c2 (https://elf-council-frontend-staging-2wessao57-element-finance.vercel.app).

File: src/ui/proposals/GSCMember.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  const { account, provider } = props;
+
+  const formattedAddress = useFormattedWalletAddress(account, provider);
+  return (
+    <Button variant={ButtonVariant.OUTLINE_WHITE}>
+      <div className="flex w-full items-center overflow-hidden">
+        <WalletJazzicon
+          size={28}
+          account={account}
+          className="mr-4 flex-shrink-0 flex-grow-0"
```
Comment (ryangoree) [inline src/ui/proposals/GSCMember.tsx:22]: `flex-shrink-0` and `flex-grow-0` can be removed

File: src/ui/proposals/GSCMember.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        />
+        <div className="max-w-0 flex-shrink text-sm font-thin">
```
Comment (ryangoree) [inline src/ui/proposals/GSCMember.tsx:24]: Why `max-w-0`? `max-w-0` and `flex-shrink` can be removed.

**Review by ryangoree — APPROVED** [review]
LGTM.

I know this is still a WIP, but I went through some of the components pretty thoroughly to find opportunities to make the markup and classes more efficient.

I didn't do this on all of them, though since it's still a WIP and I've spent enough time on it already 😅.

File: src/ui/gsc/GSCMemberProfileRow.tsx [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+function GSCMemberProfileRow(props: GSCMemberProfileRowProps): ReactElement {
+  const {
+    provider,
+    selected = false,
+    highlightSelected = false,
+    delegate,
+    delegateButton,
+    kickButton,
+    profileActionButton,
+  } = props;
```
[the remark this answers is not in the document]
Comment (sentilesdal) [reply src/ui/gsc/GSCMemberProfileRow.tsx:37]: makes no difference to me, traditionally we've done it in the body.

[the remark this answers is not in the document]
Comment (sentilesdal) [reply src/ui/gsc/GSCMemberProfileRow.tsx:37]: i'm gonna leave this.  i looked around and most do it in the body so i'm sticking with the pattern.

File: src/ui/gsc/GSCMemberProfileRow.tsx [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+  const formattedDelegateName =
+    delegate.ensName ||
+    delegate.commonwealthName ||
+    delegate.name ||
+    formatWalletAddress(delegate.address);
```
[the remark this answers is not in the document]
Comment (sentilesdal) [reply src/ui/gsc/GSCMemberProfileRow.tsx:43]: ah nice, didnt know we had that.  ur suggestion looks good.

[the remark this answers is not in the document]
Comment (sentilesdal) [reply src/ui/gsc/GSCMemberProfileRow.tsx:43]: I changed useEnsName to useENSName :).  The code that's there already assumes there is an ens name.  Not sure why wee need the useENSName hook?

File: src/ui/gsc/GSCMemberProfileRow.tsx [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+      <div
+        className={classNames(
+          "grid grid-cols-10 items-center justify-between rounded-xl bg-white py-3 px-4",
+          {
+            "!bg-votingGreen": highlightSelected && selected,
+          },
+        )}
+      >
+        {/* Name */}
+        <div className="col-span-7 mr-4 items-start truncate text-left lg:col-span-4">
```
[the remark this answers is not in the document]
Comment (sentilesdal) [reply src/ui/gsc/GSCMemberProfileRow.tsx:58]: yeah prolly a lot of cruft from where I copied :P

File: src/ui/gsc/GSCMemberProfileRow.tsx [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+          <div className="flex flex-col">
+            <div
+              className={classNames(
+                "flex items-center font-bold text-principalRoyalBlue",
+              )}
```
[the remark this answers is not in the document]
Comment (sentilesdal) [reply src/ui/gsc/GSCMemberProfileRow.tsx:63]: hah dunno

File: src/ui/gsc/GSCMemberProfileRow.tsx [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+            <div className="lg:hidden">
+              <span
+                className={
+                  highlightSelected && selected
+                    ? "text-gray-400"
+                    : "text-blueGrey"
+                }
+              >
+                <NumDelegatedVotes account={delegate.address} />
+              </span>
```
[the remark this answers is not in the document]
Comment (sentilesdal) [reply src/ui/gsc/GSCMemberProfileRow.tsx:94]: Yeah will do.  I'll clean a lot of this up as I start working in here too.

[the remark this answers is not in the document]
Comment (sentilesdal) [reply src/ui/gsc/GSCMemberProfileRow.tsx:94]: I cleaned this up a bit, gotta remember to do this to DelegateProfileRow too where I stole this from.

File: src/ui/gsc/GSCMemberProfileRow.tsx [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+
+        {/* Voting Power */}
+        <div className="col-span-2 ml-auto mr-10 hidden lg:block">
+          <span
+            className={
+              highlightSelected && selected ? "text-gray-400" : "text-blueGrey"
+            }
+          >
+            <NumDelegatedVotes account={delegate.address} />
+          </span>
```
[the remark this answers is not in the document]
Comment (sentilesdal) [reply src/ui/gsc/GSCMemberProfileRow.tsx:107]: yup changed it and cleaned up div soup

File: src/ui/gsc/GSCOverviewPage.tsx [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+  const selectedDelegate = delegates[0].address;
+  const delegateAddressOnChain = delegates[0].address;
+
+  return (
+    <div className="h-full w-full space-y-6 xl:max-w-[1024px]">
+      <Head>
+        <title>{t`GSCOverview | Element Council Protocol`}</title>
+      </Head>
+
+      <div className="px-8 py-1">
```
[the remark this answers is not in the document]
Comment (sentilesdal) [reply src/ui/gsc/GSCOverviewPage.tsx:51]: removed div

File: src/ui/gsc/GSCOverviewPage.tsx [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+        <H1 className="text-center text-principalRoyalBlue">
+          {t`Governance GSCOverview`}
+        </H1>
+      </div>
+      <div className="flex w-full grid-cols-2 flex-col justify-center space-y-6 xl:flex-row xl:space-x-6 xl:space-y-0">
```
[the remark this answers is not in the document]
Comment (sentilesdal) [reply src/ui/gsc/GSCOverviewPage.tsx:56]: removed

File: src/ui/gsc/GSCOverviewPage.tsx [hunk 11]
```diff
@@ -0,0 +1,204 @@
...
+        <div className="flex w-full flex-col gap-6">
```
[the remark this answers is not in the document]
Comment (sentilesdal) [reply src/ui/gsc/GSCOverviewPage.tsx:57]: removed the extra div.  actually not sure why there's flex and grid here 🤔

File: src/ui/gsc/GSCOverviewPage.tsx [hunk 12]
```diff
@@ -0,0 +1,204 @@
...
+          <H2 className="text-principalRoyalBlue">{t`Council Assembly`}</H2>
+          {/* Members */}
+          <ul
+            // 392px exactly matches 5 rows of the list
+            className="flex h-[40vh] min-h-[392px] w-full flex-col gap-y-2 overflow-y-scroll pr-1"
```
[the remark this answers is not in the document]
Comment (sentilesdal) [reply src/ui/gsc/GSCOverviewPage.tsx:62]: again, not sure :P

File: src/ui/gsc/GSCOverviewPage.tsx [hunk 13]
```diff
@@ -0,0 +1,204 @@
...
+              // TODO: Remove -${idx} for production since addresses are always unique
+              return (
+                <li className="w-full" key={`${member.address}-${idx}}`}>
+                  <GSCMemberProfileRow
+                    provider={provider}
+                    selected={selected}
+                    delegate={member}
+                    kickButton={
+                      member.address === members[4].address ? (
+                        <Button className="w-full bg-red-500 text-center">
```
[the remark this answers is not in the document]
Comment (sentilesdal) [reply src/ui/gsc/GSCOverviewPage.tsx:82]: I'll add another variant instead

[the remark this answers is not in the document]
Comment (sentilesdal) [reply src/ui/gsc/GSCOverviewPage.tsx:82]: added DANGER variant

File: src/ui/gsc/GSCOverviewPage.tsx [hunk 14]
```diff
@@ -0,0 +1,204 @@
...
+  currentlyDelegated,
+}: ChangeDelegateButtonProps): ReactElement {
+  return (
+    <React.Fragment>
+      {currentlyDelegated ? (
+        <Tag
+          intent={Intent.SUCCESS}
+          className={classNames(tagClassName, "w-full shadow")}
+        >
+          <div className="text-center text-base font-bold">{t`Delegated`}</div>
```
[the remark this answers is not in the document]
Comment (sentilesdal) [reply src/ui/gsc/GSCOverviewPage.tsx:189]: hmm, not sure why there's a div at all.. gonna remove it and add classes above

File: src/ui/proposals/GSCMember.tsx (hunk 2 above)
Comment (sentilesdal) [reply src/ui/proposals/GSCMember.tsx:24]: there was weird spacing issues, fixed and removed.

File: src/ui/gsc/GSCOverviewPage.tsx [hunk 15]
```diff
@@ -0,0 +1,204 @@
...
+                </li>
+              );
+            })}
+          </ul>
+
+          <H2 className="text-principalRoyalBlue">{t`Candidates`}</H2>
+          {/* Delegates */}
+          <ul
+            // 392px exactly matches 5 rows of the list
+            className="flex h-[40vh] min-h-[392px] w-full flex-col gap-y-2 overflow-y-scroll pr-1"
```
[the remark this answers is not in the document]
Comment (sentilesdal) [reply src/ui/gsc/GSCOverviewPage.tsx:116]: I removed all these, they were specific to the delegation page.

File: src/ui/gsc/GSCOverviewPage.tsx (hunk 12 above)
[the remark this answers is not in the document]
Comment (sentilesdal) [reply src/ui/gsc/GSCOverviewPage.tsx:62]: removed all these, they were copy pasta

**Pull request merged**